qiita_db/metadata_template/test/test_prep_template.py

```diff
@@ -802,58 +802,27 @@ def test_create_error_cleanup(self):
-
-        # The relevant rows to prep_template_sample have been added.
-        obs = self.conn_handler.execute_fetchall(
-            "SELECT * FROM qiita.prep_template_sample "
-            "WHERE prep_template_id=%s", (new_id,))
-        # prep_template_id, sample_id, center_name,
-        # center_project_name, emp_status_id
-        exp = [[new_id, '1.SKB8.640193'],
-               [new_id, '1.SKD8.640184'],
-               [new_id, '1.SKB7.640196']]
-        self.assertItemsEqual(obs, exp)
-
-        # The relevant rows have been added to the prep_columns table
-        obs = self.conn_handler.execute_fetchall(
-            "SELECT * FROM qiita.prep_columns WHERE prep_template_id=2")
-        # prep_template_id, column_name, column_type
-        exp = [[new_id, 'str_column', 'varchar'],
-               [new_id, 'ebi_submission_accession', 'varchar'],
-               [new_id, 'run_prefix', 'varchar'],
-               [new_id, 'barcode', 'varchar'],
-               [new_id, 'primer', 'varchar'],
-               [new_id, 'platform', 'varchar'],
-               [new_id, 'instrument_model', 'varchar'],
-               [new_id, 'experiment_design_description', 'varchar'],
-               [new_id, 'library_construction_protocol', 'varchar'],
-               [new_id, 'center_name', 'varchar'],
-               [new_id, 'center_project_name', 'varchar'],
-               [new_id, 'emp_status', 'varchar']]
-        self.assertItemsEqual(obs, exp)
-
-        # The new table exists
-        self.assertTrue(exists_table("prep_%s" % new_id))
-
-        # The new table hosts the correct values
-        obs = [dict(o) for o in self.conn_handler.execute_fetchall(
-            "SELECT * FROM qiita.prep_%s" % new_id)]
-
-        exp = [{'sample_id': '1.SKB7.640196',
+        self.assertEqual(pt.data_type(), self.data_type)
+        self.assertEqual(pt.data_type(ret_id=True), self.data_type_id)
+        self.assertEqual(pt.raw_data, None)
+        self.assertEqual(pt.preprocessed_data, [])
+        self.assertEqual(pt.preprocessing_status, 'not_preprocessed')
+        self.assertEqual(pt.investigation_type, None)
+        self.assertEqual(pt.study_id, self.test_study.id)
+        self.assertEqual(pt.status, "sandbox")
+        exp_sample_ids = {'%s.SKB8.640193' % self.test_study.id,
+                          '%s.SKD8.640184' % self.test_study.id,
+                          '%s.SKB7.640196' % self.test_study.id}
+        self.assertEqual(pt._get_sample_ids(), exp_sample_ids)
+        self.assertEqual(len(pt), 3)
+        exp_categories = {'str_column', 'ebi_submission_accession',
+                          'run_prefix', 'barcode', 'primer', 'platform',
+                          'instrument_model', 'experiment_design_description',
+                          'library_construction_protocol', 'center_name',
+                          'center_project_name', 'emp_status'}
+        self.assertEqual(set(pt.categories()), exp_categories)
```

Comment:
Good point. I always forget the existence of the `assertItemsEqual`. I guess a sticky note in my desk will help :smile: